improver_tests/calibration/reliability_calibration/test_ManipulateReliabilityTable.py

```diff
@@ -186,6 +186,26 @@ def test_non_monotonic(self):
         )
         assert_array_equal(result.data, expected_result)
 
+    def test_non_monotonic_lower_forecast_count_on_left(self):
+        """Test enforcement of monotonicity for observation frequency."""
+        obs_count = np.array([0, 750, 500, 1000, 750], dtype=np.float32)
+        forecast_count = np.array([500, 1000, 1000, 1000, 1000], dtype=np.float32)
+        expected_result = np.array([0, 500, 500, 750, 750], dtype=np.float32)
+        result = Plugin()._assume_constant_observation_frequency(
+            obs_count, forecast_count
+        )
+        assert_array_equal(result.data, expected_result)
+
+    def test_non_monotonic_higher_forecast_count_on_left(self):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6088429)